slsDetectorSoftware/src/Module.h

```diff
@@ -988,36 +979,14 @@ class Module : public virtual slsDetectorDefs {
      */
     void setTransmissionDelayRight(int value);
 
-    /**
-     * Sets the additional json header\sa sharedSlsDetector
-     * @param jsonheader additional json header
-     */
-    void setAdditionalJsonHeader(const std::string &jsonheader);
-
-    /**
-     * Returns the additional json header \sa sharedSlsDetector
-     * @returns the additional json header, returns "none" if default setting
-     * and no custom ip set
-     */
-    std::string getAdditionalJsonHeader();
-
-    /**
-     * Sets the value for the additional json header parameter if found, else
-     * append it
-     * @param key additional json header parameter
-     * @param value additional json header parameter value (cannot be empty)
-     * @returns the additional json header parameter value,
-     * empty if no parameter found in additional json header
-     */
-    std::string setAdditionalJsonParameter(const std::string &key,
-                                           const std::string &value);
+    /** empty vector deletes entire additional json header */
+    void setAdditionalJsonHeader(const std::map<std::string, std::string> jsonHeader);
```

Comment:
Since we use map as const we could probably use a reference

Reply:
Yes, will fix this

Reply:
done